### PR TITLE
Improve settings UI

### LINKED ISSUE #539: GD-539: Settings UI help strings should not include internal data
**Describe What**
Currently, the settings UI includes internal data used by the code to know what possible options a settings has. This is a consequence of the fact that `ProjectSettings`  does not give an easy way to store that information. But the code should take care to remove those details prior to showing the help to the user.

**Describe Why**
Seeing internal details included in the help string is not useful to the user, and makes the help worse than it could be. They can already see the same information presented as a dropdown, and the help should be as concise as is possible.

### LINKED ISSUE #540: GD-540: Improve user-facing help strings in settings UI
**Describe What**
Improve the settings help by removing unnecessarily verbose and repetitive elements, things which are already obvious from the UI, and ironing out inconsistencies. Make sure the help strings are understandable on their own; currently, some of them are really hard to understand without consulting the docs.

**Describe Why**
Help text is a critical part of the UI, and the more concise and obviously applicable to the UI it is, the more value it has. It should give information the user doesn't have, and not repeat things they already know from seeing the UI. These are pretty well-established principles of good UI text design.

### LINKED ISSUE #541: GD-541: Settings dialogue reserves way more horizontal space than it needs, leading to huge scrollbars
**Describe What**
Currently the settings dialogue uses a very crude estimation of what horizontal space labels need, and also has `custom_minimum_size` hardcoded in the scene. This makes it so that it requests a lot more space than it actually needs from the scrolling container, and that makes huge scrollbars show up unnecessarily. It also makes the components wider than they need to be, so the user has to scroll a lot.

**Describe Why**
It's not great UI if the user has to scroll to see things.

### PR DESCRIPTION
**Pull request by mathrick**
Hi, this PR improves the settings dialogue UI, making it significantly more compact, and IMHO, also easier to understand and nicer to look at.

Before:
gdunit-before

After:
gdunit-after

The changes are split into three commits

1. Remove the value sets of settings from the user-facing help strings. Fixes #539 
2. Improve the wording of the help strings themselves (more detailed description in the commit message). Fixes #540 
3. Improved sizing logic for the grid views displaying the settings, meaning less space is wasted, and huge amounts of unused space don't make the scrollbars huge unnecessarily. Not every tab in the dialogue is currently able to fit in without scrolling on my laptop, but some of them are, and it's generally much nicer. Fixes #541

I did not make any changes to the settings themselves, everything is as it was before and split across tabs in the same way. I did a full test suite run locally, and everything works aside from the C# tests, which I did not run, as I'm not using Godot Mono.

### CODE REVIEW

**Comment by MikeSchulze** [comment]
Hello, thank you for your contribution to this project.

Please be so kind and create another refactoring ticket detailing what you want to achieve and link it to the PR.
It should point out existing problems and possible solutions. 
It would be good to have a before/after screenshot in the PR, which can then also be used later for the release notes.
I'll take a look at your changes later today.

File: addons/gdUnit4/src/core/GdUnitSettings.gd [hunk 1]
```diff
@@ -280,20 +283,29 @@ static func list_settings(category :String) -> Array[GdUnitProperty]:
...
+	var matches := regex.search_all(split_value[1])
 	if matches.is_empty():
 		return PackedStringArray()
-	var values :String =  matches[0].get_string(1)
+	var values: String = matches[0].get_string(1)
+	var help: String
 	return values.replacen(" ", "").replacen("\"", "").split(",", false)
 
 
+static func cleanup_help(value :String) -> String:
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/core/GdUnitSettings.gd:305]:
```suggestion
static func extract_help_text(value :String) -> String:
```

File: addons/gdUnit4/src/ui/settings/GdUnitSettingsDialog.tscn [hunk 2]
```diff
@@ -117,7 +117,7 @@ shadow_offset = Vector2(10, 10)
 [node name="Control" type="Window"]
 disable_3d = true
 gui_embed_subwindows = true
-title = "GdUnitSettings"
+title = "gdUnit4 Settings"
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/ui/settings/GdUnitSettingsDialog.tscn:120]:
```suggestion
title = "GdUnit4 Settings"
```

File: addons/gdUnit4/src/core/GdUnitSettings.gd [hunk 3]
```diff
@@ -280,20 +283,29 @@ static func list_settings(category :String) -> Array[GdUnitProperty]:
...
+
 	var regex := RegEx.new()
 	regex.compile("\\[(.+)\\]")
-	var matches := regex.search_all(value)
+	var matches := regex.search_all(split_value[1])
 	if matches.is_empty():
 		return PackedStringArray()
-	var values :String =  matches[0].get_string(1)
+	var values: String = matches[0].get_string(1)
+	var help: String
```
Comment (MikeSchulze) [inline addons/gdUnit4/src/core/GdUnitSettings.gd:301]:
```suggestion
```

**Review by MikeSchulze — CHANGES_REQUESTED** [review]
I have checked the changes and looks good.
A have only find minor things, please check my comments.

In addition, the actual dialog size can be shrinking because the properties are now more compact and well sized.
I tried with using `reset_size()` after component creation but without success, maybe you can fix this too.

Great Job, Thanks

File: addons/gdUnit4/src/core/GdUnitSettings.gd (hunk 1 above)
Comment (mathrick) [reply addons/gdUnit4/src/core/GdUnitSettings.gd]: Done

File: addons/gdUnit4/src/ui/settings/GdUnitSettingsDialog.tscn (hunk 2 above)
Comment (mathrick) [reply addons/gdUnit4/src/ui/settings/GdUnitSettingsDialog.tscn:120]: Done

File: addons/gdUnit4/src/core/GdUnitSettings.gd (hunk 3 above)
Comment (mathrick) [reply addons/gdUnit4/src/core/GdUnitSettings.gd]: Oops, leftover code I should have removed, fixed

**Comment by mathrick** [comment]
> Please be so kind and create another refactoring ticket detailing what you want to achieve and link it to the PR. It should point out existing problems and possible solutions. It would be good to have a before/after screenshot in the PR, which can then also be used later for the release notes. I'll take a look at your changes later today.

I made the refactoring tickets, I'll take some screenshots and attach them later today.

> In addition, the actual dialog size can be shrinking because the properties are now more compact and well sized. I tried with using `reset_size()` after component creation but without success, maybe you can fix this too.

Yeah, I ran into that too :(. It seems that no matter what I do, the actual dialogue size won't budge. I don't really understand the way Godot windows work, so I had little luck getting it to do what I want. I tried changing the size in the saved scene, but that value is apparently just completely ignored. The thing that displays on my screen has a completely different shape than what is shown in the 2D editor preview.

**Comment by mathrick** [comment]
Alright, here are some screenshots. Do these work for you?

Before:
gdunit-before

After:
gdunit-after

**Review by MikeSchulze — APPROVED** [review]

**Comment by MikeSchulze** [comment]
Great job, thanks for contribution to the project.

**Pull request merged**